src/prefect/core/task.py

```diff
@@ -315,6 +319,7 @@ def __init__(
         tags: Iterable[str] = None,
         max_retries: int = None,
         retry_delay: timedelta = None,
+        retry_on: Iterable[Type[Exception]] = None,
```

Comment:
Should we accept a non-iterable `Exception` too?

```suggestion
        retry_on: Union[Type[Exception], Iterable[Type[Exception]]] = None,
```

I imagine there'd be a lot more people checking a single exception rather than multiple.

Reply:
Sure, want to push a commit that adds support?